Lab1/Lab1/Services/IOSystem/Readers/CsvReader.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Lab1.Services.Exceptions;
+using Lab1.Models;
+using Lab1.Services.Interfaces;
+using Lab1.Services.Parsers;
+using Lab1.Services.Validation;
+
+namespace Lab1.Services.IOSystem.Readers
+{
+    public class CsvReader : IReader<Student>
+    {
+        public IEnumerable<Student> Read(string path, out List<string> fieldNames)
+        {
+            path.CsvPathValidation();
+            path.CheckFileExistance();
+            
+            List<Student> students = new List<Student>();
+            bool firstString = true;
+            fieldNames = new List<string>();
+
+            using (StreamReader streamReader = new StreamReader(path))
+            {
+                while (!streamReader.EndOfStream)
+                {
+                    string pattern = streamReader.ReadLine();
+
+                    if (firstString)
+                    {
+                        fieldNames = pattern.Split(',', ';').ToList();
```

Comment:
It's used very often and not in single class - make it reusable.

Reply:
Make method ParsePattern in class StudentsCsvParser